src/usr/local/www/diag_packet_capture.php

```diff
@@ -512,7 +514,17 @@ function hostport_array_fixer($items,$type) {
 	}
 
 	if (in_array($proto, $protos)) {
-		$matches[] = fixup_not(str_replace('carp', 'proto 112', $proto));
+		if ($proto == 'carp') {
```

Comment:
I'm not fond of this structure. It looks like it's going to get even uglier if anything else is added. If it is necessary, it should be turned into a switch based on the contents of `$proto` with the default action being to not do a `str_replace()`. Or perhaps the switch only alters the contents of `$proto` and then after that block, gets `fuxup_not` done and added to `$matches` just once at the end, rather than repeating it so much.

Reply:
fixed

extend fixup_not() for !ospf, 
because 'tcpdump -i vmx0 ip and ! ospf' is incorrect, 
should be  'tcpdump -i vmx0 ip and ! proto ospf'